src/Components/OATPropertyEditor/TemplateList.tsx

```diff
@@ -143,6 +143,39 @@ export const TemplateList = ({
         });
     };
 
+    const handlePropertyListAddition = (item) => {
+        if (model) {
+            const newModel = deepCopy(model);
+            newModel[propertiesKeyName].push(item);
+            dispatch({
+                type: SET_OAT_PROPERTY_EDITOR_MODEL,
+                payload: newModel
+            });
+        }
+    };
+
+    const moveItemUpOnTemplateList = (index: number) => {
+        const newTemplate = deepCopy(templates);
+        const item = newTemplate[index];
+        newTemplate.splice(index, 1);
+        newTemplate.splice(index - 1, 0, item);
+        dispatch({
+            type: SET_OAT_TEMPLATES,
+            payload: newTemplate
+        });
+    };
+
+    const moveItempDownOnTemplateList = (index: number) => {
+        const newTemplate = deepCopy(templates);
+        const item = newTemplate[index];
+        newTemplate.splice(index, 1);
+        newTemplate.splice(index + 1, 0, item);
+        dispatch({
+            type: SET_OAT_TEMPLATES,
+            payload: newTemplate
+        });
+    };
```

Comment:
This and the prior function are exactly the same apart from the `+1` and `-1`. Maybe we could get some code reuse here?